### PR TITLE
feat(payment): PAYPAL-0000 updated braintree paypal checkout button strategy to prepare strategies separation

### PR DESCRIPTION
**Pull request by serhii-tkachenko**
## What?
- updated braintree paypal checkout button strategy;
- added some updates to braintree checkout sdk;
- updated braintree types and interfaces;
- cleaned some code and fixed some tests;

## Why?
To prepare braintree checkout buttons strategies separation.

## Testing / Proof
Unit tests
Manual tests

US:
Screenshot 2022-06-06 at 12 18 42

Other countries:
Screenshot 2022-06-06 at 12 16 18

### CODE REVIEW

File: packages/core/src/checkout-buttons/strategies/braintree/get-valid-button-style.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    };
+
+    return omitBy(validStyles, isNil);
+}
+
+function validateHeight(height?: number): number | undefined {
+    const minHeight = 25;
+    const maxHeight = 55;
+
+    if (typeof height !== 'number') {
```
Comment (animesh1987) [inline packages/core/src/checkout-buttons/strategies/braintree/get-valid-button-style.ts:25]: Can we just return maxHeight over here?

File: packages/core/src/checkout-buttons/strategies/braintree/get-valid-button-style.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        return undefined;
+    }
+
+    return (height < minHeight && minHeight)
```
Comment (animesh1987) [inline packages/core/src/checkout-buttons/strategies/braintree/get-valid-button-style.ts:29]: Can we use if/else or switch here.

File: packages/core/src/checkout-buttons/strategies/braintree/map-to-legacy-shipping-address.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { LegacyAddress } from '../../../address';
+import { BraintreeDetails } from '../../../payment/strategies/braintree';
+
+export default function mapToLegacyShippingAddress(details: BraintreeDetails): Partial<LegacyAddress> {
+    const { email, phone, shippingAddress } = details;
+
+    const recipientName = shippingAddress?.recipientName || '';
```
Comment (animesh1987) [inline packages/core/src/checkout-buttons/strategies/braintree/map-to-legacy-shipping-address.ts:7]: Can we add a check and not use empty string as a fallback?

File: packages/core/src/checkout-buttons/strategies/braintree/get-valid-button-style.ts (hunk 2 above)
Comment (serhii-tkachenko) [reply packages/core/src/checkout-buttons/strategies/braintree/get-valid-button-style.ts:29]: Yes, it's a bit tricky to read

File: packages/core/src/checkout-buttons/strategies/braintree/get-valid-button-style.ts (hunk 1 above)
Comment (serhii-tkachenko) [reply packages/core/src/checkout-buttons/strategies/braintree/get-valid-button-style.ts:25]: I think we can

**Review by animesh1987 — APPROVED** [review]

**Pull request merged**